src/frontend/src/lib/api/backend.api.ts

```diff
@@ -8,12 +9,29 @@ import type {
 	UserToken
 } from '$declarations/backend/backend.did';
 import { getBackendActor } from '$lib/actors/actors.ic';
-import type { EthAddress } from '$lib/types/address';
+import type { BtcAddress, EthAddress } from '$lib/types/address';
 import type { OptionIdentity } from '$lib/types/identity';
 import type { Identity } from '@dfinity/agent';
 import type { Principal } from '@dfinity/principal';
 import { toNullable, type QueryParams } from '@dfinity/utils';
 
+const getBtcAddress = async ({
+	identity,
+	network
+}: {
+	identity: OptionIdentity;
+	network: BitcoinNetwork;
+}): Promise<EthAddress> => {
+	const { caller_btc_address } = await getBackendActor({ identity });
+	return caller_btc_address(network);
+};
+
+export const getBtcAddressMainnet = async (identity: OptionIdentity): Promise<BtcAddress> =>
```

Comment:
Somehow, I remember commenting that I prefer not to have two distinct functions at the API level, which, in my opinion, is only meant to serve as a sort of proxy. Also commenting that those kind of logic can be applied at the services level.

However, I cannot recall where or what your response was—specifically, why we have those two distinct functions. Could you remind me where this discussion or these comments took place?

Reply:
> Maybe I understood wrong. You meant to have just one getBtcAddress and then we map the networks when we are using it?

Yes 👍

We can discuss it and I'm not 100% I was consistent but, I see the `api` layer as a dumb proxy layer. Ideally, I would even generate it automatically, like it get's value, get the actor, pass those to the actor and that's it. You know what I mean?

Reply:
yes, I think I understand eheh! I read through the entire module and indeed it's that. Maybe you wanted to reduce the repetition of code (each time call `getBackendActor`) or concentrate it somewhere, so that if something changed on the backend you could easily change it from there.

In the meanwhile, I will remove the two repeated functions.